test/handler/test_vint_handler.vader

```diff
@@ -10,40 +10,46 @@ Execute(The vint handler should parse error messages correctly):
   \   {
   \     'lnum': 1,
   \     'col': 1,
+  \     'filename': 'gcc.vim',
   \     'text': 'Use scriptencoding when multibyte char exists (see :help :script encoding)',
   \     'type': 'W',
   \   },
   \   {
   \     'lnum': 3,
   \     'col': 17,
+  \     'filename': 'gcc.vim',
   \     'end_col': 18,
   \     'text': 'Use robust operators ''==#'' or ''==?'' instead of ''=='' (see Google VimScript Style Guide (Matching))',
   \     'type': 'W',
   \   },
   \   {
   \     'lnum': 3,
   \     'col': 8,
+  \     'filename': 'gcc.vim',
   \     'end_col': 15,
   \     'text': 'Make the scope explicit like ''l:filename'' (see Anti-pattern of vimrc (Scope of identifier))',
   \     'type': 'W',
   \   },
   \   {
   \     'lnum': 7,
   \     'col': 8,
+  \     'filename': 'gcc.vim',
```

Comment:
It's hard to tell with test data, but try to set absolute paths if you can. The real output might actually use absolute paths.

Reply:
With clangcheck at least, the output contains the absolute path so there's no need to worry too much.

Maybe vint outputs relative paths, but then I don't know if we should expect the linters to have options to output absolute path, or if we should add another function in the gcc handler script to find the absolute path (maybe `expand(filename:p)` works ?)

Reply:
I think Vint only works with a single file, I wouldn't worry about Vint.

Reply:
For most linters, if we needed to work out the absolute path ourselves, the way to do that is to determine what directory the paths are actually relative to, and then resolve the relative paths based on that path.

Tools which output absolute paths are better.

Reply:
The paths are given as relative to the location where the linter is launched, that is the current working directory of vim.

Even if I `:lcd` to another directory (and the loclist still shows the same relative path to the starting vim working directory), jumping to other files works as intended. I think it's safe to assume that vim handles these changes on its own

once again, the output is messy but please don't mind the errors, it's from clang_complete that I need to learn to configure properly https://asciinema.org/a/0DchjjNMt8xxuiuyEqOhS9oqP

Reply:
I wouldn't assume that.

The handlers for the errors can be executed from some buffer other than the buffer the command was run from. That's why `getcwd()` is blacklisted in the codebase. It can give the wrong output.